### ISSUE TITLE
Rich notifications on iOS 13

### ISSUE THREAD

**Issue by BadgerBadgerAndFox** [hjdhjd/homebridge-unifi-protect2#4]
Due to changes in iOS 13 rich notifications no longer work if the motion sensor is not part of the camera (accessory of). This has been resolved through a pull request in homebridge-camera-ffmpeg initiated by NorthernMan54 (https://github.com/KhaosT/homebridge-camera-ffmpeg/issues/332)
@hjdhjd Would it be possible to implement the recent updates in homebridge-camera-ffmpeg to facilitate this? I have merged the code into index.js and it seems to be working. Alternately as this solution is specific to Unifi Protect would it be possible to merge in the motion sensor capability from homebridge-unifi-protect-motion-sensors?

**Comment by loop2** [comment]
@BadgerBadgerAndFox can you explain how you get the motion sensor please? thanks

**Comment by BadgerBadgerAndFox** [comment]
@loop2 you’ll need to manually merge in the code changes to index.js until it is included. You’ll need a basic understanding of js to make sure you don’t break it.

**Comment by hjdhjd** [comment]
@BadgerBadgerAndFox I saw the issue related to the post you mention. Here's the rub...when you tie yourself, as I did, to homebridge-camera-ffmpeg, you're reliant on the upstream to fix things. Temporary workarounds aren't a great idea here because they're unmaintanable. So I'll wait on the upstream for this one. I don't think it'll be more than a couple of more weeks at most.

Incorporating similar functionality to what's in homebridge-unifi-protect-motion-sensors is something I've looked into, but need dedicated time to sort through. If someone wants to take on that challenge, let me know. :)

One of the concerns I have about the motion sensor API in Protect is that there really isn't one. Most of the solutions are tapping into the log stream and parsing the content there. It's inelegant at best, and likely a performance nightmare I would expect. If someone's found a better way to **directly get at motion sensor data via API**, I'm all for it...point me in the right direction. Directly being the operative term.

Let's revisit in a couple of weeks...

**Comment by ryanschmidt** [comment]
@hjdhjd any updates on the upstream update? It would be great to have iOS Push Notifications again. Thanks for the great plugin!

**Comment by hjdhjd** [comment]
If the upstream implemented the ffmpeg update, you need to update homebridge-camera-ffmpeg and you should have most of what you need.

Here’s the thing that remains a struggle on motion sensors - it’s a moving target. Ubiquiti is clearly making significant changes in this space right now with the addition of the new / enhanced motion sensor algorithms, etc. - as the APIs stabilize and one can get this going in a useful way, I’d love to implement it in this package.

Most of what I’ve seen in the homebridge world makes me cringe as it’s remained a kludge of sorts...read the entire log stream and try to make sense of it for the events you care about. If you have a moderate-sized environment or larger, that’s going to be a performance pain point.

Honestly...not to pontificate on wishlists, but hey it’s the middle of the night for me...what we really need is a properly documented and implemented Ubiquiti API that we can tap into. They’ve been pretty clear that it’s an unlikely possibility but one can hope.

TLDR: if/when there’s an API-driven way to do motion sensing, I’d like to implement it. I haven’t seen it yet...but Ubiquiti is making a lot of updates to that part of Protect these days, so perhaps in the coming months...
